pkg/controllers/state/cluster.go

```diff
@@ -53,7 +53,9 @@ type Cluster struct {
 	nodeNameToProviderID      map[string]string               // node name -> provider id
```

Comment:
Are there any tests that we should add to cluster state around this new function that you've introduced here?